### PR TITLE
Allow Sidebar Menu search using text from Item Name of a record

### PR DESCRIPTION
**Pull request by creamdory**
From PR in frappe repo https://github.com/frappe/frappe/pull/3798#pullrequestreview-52562113 as requested changes.

For web pages in Menu Sidebars, the common column for doctypes with "items" child table include item name's summary as below:

item names

It would be best to allow having this column to look at when using the page's **Search Bar.**

item name filter

This is provided that "items" field is included in the doctype's search fields:

search field

### CODE REVIEW

**Comment by rmehta** [comment]
- [ ] Test Case
- [ ] Screenshots

**Comment by creamdory** [comment]
Edited the code to automatically allow item_name search for doctypes with 'items' field.

File: erpnext/controllers/website_list_for_contact.py [hunk 1]
```diff
@@ -45,17 +44,42 @@ def get_transaction_list(doctype, txt=None, filters=None, limit_start=0, limit_p
...
 				ignore_permissions=True,
 				order_by = "modified desc"))
 		else:
 			return []
 
-	return post_process(doctype, get_list(doctype, txt, filters, limit_start, limit_page_length,
+	return post_process(doctype, get_trans_data_list(doctype, txt, filters, limit_start, limit_page_length,
 		fields="name", order_by = "modified desc"))
 
+def get_trans_data_list(doctype, txt, filters, limit_start, limit_page_length=20, ignore_permissions=False,fields=None, order_by=None):
```
Comment (rmehta) [inline erpnext/controllers/website_list_for_contact.py]: no shortnames like `trans_data` - use `get_list_for_transactions`

**Review by rmehta — CHANGES_REQUESTED** [review]
Also write a simple test case to check this function works!

File: erpnext/controllers/website_list_for_contact.py (hunk 1 above)
Comment (creamdory) [inline erpnext/controllers/website_list_for_contact.py]: Ok sure @rmehta , I'll make the changes and make a simple test case. Thanks! I'll make update once ready.

**Comment by mbauskar** [comment]
@creamdory,

Thanks for contributing,

Please check and fix travis build issues also please add a simple test case for the PR, Closing the PR for now but you can reopen it.

**Comment by creamdory** [comment]
Sure @mbauskar , I am still preparing the test case for this. I'll let you know once ready and will re-open this one. Thanks!

**Pull request closed without merging**